udata/core/discussions/api.py

```diff
@@ -133,6 +133,26 @@ def delete(self, id):
         return '', 204
 
 
+@ns.route('/<id>/comments/<int:cidx>', endpoint='discussion_comment')
+class DiscussionCommentAPI(API):
+    '''
+    Base class for a comment in a discussion thread.
+    '''
+    @api.secure(admin_permission)
+    @api.doc('delete_discussion_comment')
+    @api.response(403, 'Not allowed to delete this comment')
+    def delete(self, id, cidx):
+        '''Delete a comment given its index'''
+        discussion = Discussion.objects.get_or_404(id=id)
+        if len(discussion.discussion) <= cidx:
+            api.abort(404, 'Comment does not exist')
+        elif len(discussion.discussion) == 1:
```

Comment:
Maybe add a condition to disallow deleting the first message (and so prevent to have an empty discussion heading followed by comments ie. admin should delete discussion instead of trying to delete the first message)

Reply:
Done, I replaced the "last comment" condition by "first comment", makes more sense.